src/Nethermind/Nethermind.Core/Collections/ArrayPoolList.cs

```diff
@@ -21,14 +22,34 @@ public sealed class ArrayPoolList<T> : IList<T>, IList, IReadOnlyList<T>, IDispo
 
     public ArrayPoolList(int capacity) : this(ArrayPool<T>.Shared, capacity) { }
 
+    public ArrayPoolList(int capacity, int count) : this(ArrayPool<T>.Shared, capacity, count) { }
+
     public ArrayPoolList(int capacity, IEnumerable<T> enumerable) : this(capacity) => this.AddRange(enumerable);
 
-    public ArrayPoolList(ArrayPool<T> arrayPool, int capacity)
+    public ArrayPoolList(ArrayPool<T> arrayPool, int capacity, int startingCount = 0)
     {
         _arrayPool = arrayPool;
-        if (capacity == 0) capacity = 16; // minimum with arraypool is 16 anyway...
-        _array = arrayPool.Rent(capacity);
+
+        if (capacity != 0)
+        {
+            _array = arrayPool.Rent(capacity);
+            for (int i = 0; i < startingCount; i++)
```

Comment:
Rather than this clear loop use span clear?

`_array.AsSpan(0, startingCount).Clear();`

Reply:
Done